code/modules/mob/living/living_fov.dm

```diff
@@ -0,0 +1,262 @@
+	if(user_living && !user_extends_eye)
+		should_apply_mask = TRUE
+	if(should_apply_mask == applied_mask)
+		return
+	if(should_apply_mask)
+		add_mask()
+	else
+		remove_mask()
+
+/// Updates the size of the FOV masks by comparing them to client view size.
+/datum/field_of_view/proc/update_fov_size()
+	var/client/parent_client = owner.client
+	if(!parent_client) //Love client volatility!!
+		return
+	var/list/view_size = getviewsize(parent_client.view)
+	if(view_size[1] == current_fov_x && view_size[2] == current_fov_y)
+		return
+	current_fov_x = BASE_FOV_MASK_X_DIMENSION
+	current_fov_y = BASE_FOV_MASK_Y_DIMENSION
+	var/matrix/new_matrix = new
+	var/x_shift = view_size[1] - current_fov_x
+	var/y_shift = view_size[2] - current_fov_y
+	var/x_scale = view_size[1] / current_fov_x
+	var/y_scale = view_size[2] / current_fov_y
+	current_fov_x = view_size[1]
+	current_fov_y = view_size[2]
+	visual_shadow.transform = blocker_mask.transform = new_matrix.Scale(x_scale, y_scale)
+	visual_shadow.transform = blocker_mask.transform = new_matrix.Translate(x_shift * 16, y_shift * 16)
+
+/// Adds the masks to the user
+/datum/field_of_view/proc/add_mask()
+	var/client/parent_client = owner.client
+	if(!parent_client) //Love client volatility!!
+		return
+	parent_client.screen += blocker_mask
+	parent_client.screen += visual_shadow
+	applied_mask = TRUE
+
+/// Removes the masks from the user
+/datum/field_of_view/proc/remove_mask()
+	var/client/parent_client = owner.client
+	if(!parent_client) //Love client volatility!!
+		return
+	parent_client.screen -= blocker_mask
+	parent_client.screen -= visual_shadow
+	applied_mask = FALSE
+
+/// When a direction of the user changes, so do the masks
+/datum/field_of_view/proc/dir_change(new_dir)
+	blocker_mask.dir = new_dir
+	visual_shadow.dir = new_dir
+
+/datum/field_of_view/proc/set_fov_angle(new_angle)
+	fov_angle = new_angle
+	blocker_mask.icon_state = "[fov_angle]"
+	visual_shadow.icon_state = "[fov_angle]_v"
+
+/// Is `observed_atom` in a mob's field of view? This takes blindness, nearsightness and FOV into consideration
+/mob/living/proc/in_fov(atom/observed_atom, ignore_self = FALSE)
+	if(ignore_self && observed_atom == src)
+		return TRUE
+	if(is_blind())
+		return FALSE
+	. = FALSE
+	var/turf/my_turf = get_turf(src) //Because being inside contents of something will cause our x,y to not be updated
+	// If turf doesn't exist, then we wouldn't get a fov check called by `play_fov_effect` or presumably other new stuff that might check this.
+	//  ^ If that case has changed and you need that check, add it.
+	var/rel_x = observed_atom.x - my_turf.x
+	var/rel_y = observed_atom.y - my_turf.y
+	if(fov_view)
+		if(rel_x >= -1 && rel_x <= 1 && rel_y >= -1 && rel_y <= 1) //Cheap way to check inside that 3x3 box around you
+			return TRUE //Also checks if both are 0 to stop division by zero
+	
+		// Converts the relative position into a 0-360 rotation
+		var/vector_len = sqrt(abs(rel_x) ** 2 + abs(rel_y) ** 2)
```

Comment:
if this is a 0-360 rotation why is the var name vector_len

Reply:
Oh my bad. I changed my method and forgot to update the comment. Will fix